dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+
+        References
+        ----------
+           [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+        """
+        ReconstModel.__init__(self, gtab)
+
+        if not callable(fit_method):
+            try:
+                self.fit_method = common_fit_methods[fit_method]
+            except KeyError:
+                raise ValueError('"' + str(fit_method) + '" is not a known fit '
+                                 'method, the fit method should either be a '
+                                 'function or one of the common fit methods')
+
+        self.design_matrix = dki_design_matrix(self.gtab)
+        self.args = args
+        self.kwargs = kwargs
+
+
+    def fit(self, data, mask=None):
+        """ Fit method of the DKI model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[-1]
+        """
+        # If a mask is provided, we will use it to access the data
+        if mask is not None:
+            # Make sure it's boolean, so that it can be used to mask
+            mask = np.array(mask, dtype=bool, copy=False)
+            data_in_mask = data[mask]
+        else:
+            data_in_mask = data
+
+        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
+                                         *self.args, **self.kwargs)
+
+        if mask is None:
+            out_shape = data.shape[:-1] + (-1, )
+            dki_params = params_in_mask.reshape(out_shape)
+        else:
+            dki_params = np.zeros(data.shape[:-1] + (27,))
+            dki_params[mask, :] = params_in_mask
+
+        return DKIFit(self, dki_params)
+
+    def predict(self, dki_params, S0=1):
+        """
+        Predict a signal for this DKI model class instance given parameters.
+
+        Parameters
+        ----------
+        dki_params : ndarray (..., 27)
+            All parameters estimated from the diffusion kurtosis model.
+            Parameters are ordered as follow:
+                1) Three diffusion tensor's eingenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+
+        gtab : a GradientTable class instance
+            The gradient table for this prediction
+
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+        """
+        return DKI_prediction(dki_params, self.gtab, S0)
+
+
+class DKIFit(TensorFit):
```

Comment:
Same as above. Please write
DiffusionKurtosisFit

Using longer and more expressive names rather than shortcuts make the code much easier to read. 
Having as much readable code as we can is one of our missions. Keep it up!